frontend/src/components/MatchingPairs/MatchingPairs.test.js

```diff
@@ -0,0 +1,156 @@
+        render(<MatchingPairs {...baseProps} sections={mockSections} setPlayerIndex={jest.fn()} />);
+        const cards = screen.getAllByRole('button'); // Assuming each card is a button, adjust if necessary
+
+        fireEvent.click(cards[0]);
+
+        await waitFor(() => expect(cards[0]).toHaveClass('turned'));
+    });
+
+    it('updates score after a match', async () => {
+    mockAxios.post.mockResolvedValueOnce({data: {score: 10}});
+    const { getByText } = render(<MatchingPairs {...baseProps} sections={mockSections} setPlayerIndex={jest.fn()} />);
+    const cards = screen.getAllByRole('button');
+
+    fireEvent.click(cards[0]);
+    fireEvent.click(cards[2]);
+
+    await waitFor( () => expect(getByText('Score: 110')).toBeInTheDocument());
+    });
+
+    xit('has a blocking overlay in-between turns', async () => {
+        mockAxios.post.mockResolvedValueOnce({data: {score: 0}});
+        render(<MatchingPairs {...baseProps} sections={mockSections} setPlayerIndex={jest.fn()} />);
+        const cards = screen.getAllByRole('button');
+
+        await waitFor(() => expect(screen.getByTestId('overlay')).toHaveStyle('display: none'));
+
+        fireEvent.click(cards[0]);
+        fireEvent.click(cards[1]);
+
+        await waitFor(() => expect(screen.getByTestId('overlay')).toHaveStyle('display: block'));
+    });
+
+    xit('calls scoreIntermediateResult after each turn', async() => {
+        const api = require('../../API');
+        const spy = jest.spyOn(api, 'scoreIntermediateResult');
+        mockAxios.post.mockResolvedValueOnce({data: {score: 10}});
+        render(<MatchingPairs {...baseProps} sections={mockSections} setPlayerIndex={jest.fn()} />);
+        const cards = screen.getAllByTestId('play-card');
+
+        fireEvent.click(cards[0]);
+        fireEvent.click(cards[2]);
+        fireEvent.click(screen.getByTestId('overlay'));
+
+        await waitFor(() => screen.getByText('Pick a card'));
+        expect(spy).toHaveBeenCalled();
+    })
+
+
+    xit('ends the game when all pairs are matched', async () => {
+        mockAxios.post.mockResolvedValue({data: {score: 10}});
+        const submitResult = jest.fn();
+        render(
+            <MatchingPairs {...baseProps} sections={mockSections} setPlayerIndex={jest.fn()} submitResult={submitResult} />
+        );
+        const cards = screen.getAllByTestId('play-card');
+
+        fireEvent.click(cards[0]);
+        fireEvent.click(cards[2]);
+        fireEvent.click(screen.getByTestId('overlay'));
+        await screen.findByText('Score: 110');
+        expect(cards[0]).toBeDisabled();
+        expect(cards[2]).toBeDisabled();
+        
+        fireEvent.click(cards[1]);
+        fireEvent.click(cards[3]);
+        fireEvent.click(screen.getByTestId('overlay'));
+        await screen.findByText('Score: 120');
+        expect(cards[1]).toBeDisabled();
+        expect(cards[3]).toBeDisabled();
+        expect(submitResult).toHaveBeenCalled();
+    });
```

Comment:
Did these tests pose problems?

Reply:
Yes, see description of the PR - `mockAxios` doesn't seem to reset properly between tests.